cluster/gce/windows/k8s-node-setup.psm1

```diff
@@ -1525,349 +1525,423 @@ function Start_Containerd {
-    try {
-      Start-Process -FilePath "$uninstall_string" -ArgumentList "/S" -Wait
-    } catch {
-      Log-Output "Exception happens during uninstall logging agent, so remove the registry key at least"
-      Remove-Item -Path "$x32_app_reg\GoogleStackdriverLoggingAgent\"
-    }
-  }
-
-  #  If we chose reboot after uninstallation, the root folder would be clean.
-  #  But since we couldn't reboot, so some files & folders would be left there, 
-  #  which could block the re-installation later on, so clean it up
-  if(Test-Path $STACKDRIVER_ROOT){
-    Remove-Item -Force -Recurse $STACKDRIVER_ROOT
-  }
+  Log-Output ("Skip: Cleanup logging agent TODO")
 }
 
-# Installs the Stackdriver logging agent according to
-# https://cloud.google.com/logging/docs/agent/installation.
-# TODO(yujuhong): Update to a newer Stackdriver agent once it is released to
-# support kubernetes metadata properly. The current version does not recognizes
-# the local resource key "logging.googleapis.com/local_resource_id", and fails
-# to label namespace, pod and container names on the logs.
+# Installs the logging agent according to https://docs.fluentbit.io/manual/installation/windows#
+# Also installs fluent bit stackdriver exporter
 function Install-LoggingAgent {
-  # Remove the existing storage.json file if it exists. This is a workaround
-  # for the bug where the logging agent cannot start up if the file is
-  # corrupted.
-  Remove-Item `
-      -Force `
-      -ErrorAction Ignore `
-      ("$STACKDRIVER_ROOT\LoggingAgent\Main\pos\winevtlog.pos\worker0\" +
-       "storage.json")
-
   if (IsLoggingAgentInstalled) {
-    # Note: we should reinstall the Stackdriver agent if $REDO_STEPS is true
+    # Note: we should reinstall the agent if $REDO_STEPS is true
     # here, but we don't know how to run the installer without it prompting
-    # when Stackdriver is already installed. We dumped the strings in the
+    # when logging agent is already installed. We dumped the strings in the
     # installer binary and searched for flags to do this but found nothing. Oh
     # well.
-    Log-Output ("Skip: Stackdriver logging agent is already installed")
+    Log-Output ("Skip: Fluentbit logging agent is already installed")
     return
   }
   
-  # After a crash, the StackdriverLogging service could be missing, but its files will still be present
+  # After a crash, the $LOGGINGAGENT_SERVICE service could be missing, but its files will still be present
   Cleanup-LoggingAgent
 
-  $url = ("https://storage.googleapis.com/gke-release/winnode/stackdriver/" +
-          "StackdriverLogging-${STACKDRIVER_VERSION}.exe")
-  $tmp_dir = 'C:\stackdriver_tmp'
-  New-Item $tmp_dir -ItemType 'directory' -Force | Out-Null
-  $installer_file = "${tmp_dir}\StackdriverLogging-${STACKDRIVER_VERSION}.exe"
-  MustDownload-File -OutFile $installer_file -URLs $url
+  Download-LoggingAgent
+  Install-LoggingAgentService
+  Remove-Item -Force -Recurse $LOGGINGAGENT_INSTALLER_DIR
```

Comment:
Can you adjust `Download-LoggingAgent` to put the file into the right place, then cleanup the INSTALLER_DIR? Let `Install-LoggingAgentService` assume that the binaries are in the right directories.

You can also add "if ShouldWrite-File ..." in `Download-LoggingAgent`, as most other download functions do in this script, so that we'll skip the download when the binary is preloaded.

Reply:
done.